### PR TITLE
Using Self in EventDescriptor

### PR DESCRIPTION
**Pull request by bithavoc**

### CODE REVIEW

File: Source/Timeline/EventDescriptor.swift [hunk 1]
```diff
@@ -11,6 +11,6 @@ public protocol EventDescriptor: AnyObject {
   var textColor: UIColor {get}
   var backgroundColor: UIColor {get}
   var editedEvent: EventDescriptor? {get set}
-  func makeEditable() -> EventDescriptor
+  func makeEditable() -> Self
```
Comment (richardtop) [inline Source/Timeline/EventDescriptor.swift]: This is a great idea.

**Comment by richardtop** [comment]
Thanks for your Pull Request. I would like to discourage from subclassing `Event` and use an own custom class conforming to the `EventDescriptor` protocol.

Please remove your changes of the `Event` which encourage subclassing (do not remove the protocol change, it's great). Then your Pull Request will be good to go!

**Comment by bithavoc** [comment]
Sure thing, however, what I wanted from Event is this:

```
  public var color = UIColor.blue {
    didSet {
      updateColors()
    }
  }
...
public weak var editedEvent: EventDescriptor? {
    didSet {
      updateColors()
    }
  }
...
  private func updateColors() {
    (editedEvent != nil) ? applyEditingColors() : applyStandardColors()
  }

  private func applyStandardColors() {
    backgroundColor = color.withAlphaComponent(0.3)
    var h: CGFloat = 0, s: CGFloat = 0, b: CGFloat = 0, a: CGFloat = 0
    color.getHue(&h, saturation: &s, brightness: &b, alpha: &a)
    textColor = UIColor(hue: h, saturation: s, brightness: b * 0.4, alpha: a)
  }

  private func applyEditingColors() {
    backgroundColor = color
    textColor = .white
  }
```

Do you have any suggestions on how to reuse that part?

**Comment by richardtop** [comment]
Could you please explain what have you changed?

**Comment by bithavoc** [comment]
Sorry, I haven't changed anything, it's just a question, I'm interested in keeping the default implementation of `color didSet` and `editedEvent didSet` that it's in the Event base class. How do you suggest I reuse that in the future?

(I'm updating the PR)

**Comment by bithavoc** [comment]
@richardtop Updated the PR, I'll raise the question in a different Issue.

**Pull request merged**